au4/src/au3wrap/internal/processinginteraction.cpp

```diff
@@ -60,3 +60,50 @@ bool ProcessingInteraction::changeClipTitle(const processing::ClipKey& clipKey,
 
     return true;
 }
+
+bool ProcessingInteraction::removeClip(const processing::ClipKey& clipKey)
+{
+    WaveTrack* waveTrack = DomAccessor::findWaveTrack(projectRef(), TrackId(clipKey.trackId));
+    IF_ASSERT_FAILED(waveTrack) {
+        return false;
+    }
+
+    std::shared_ptr<WaveClip> clip = DomAccessor::findWaveClip(waveTrack, clipKey.index);
+    IF_ASSERT_FAILED(clip) {
+        return false;
+    }
+
+    clip->Clear(clip->Start(), clip->End());
+
+    processing::ProcessingProjectPtr prj = globalContext()->currentProcessingProject();
+    prj->onClipRemoved(DomConverter::clip(waveTrack, clip.get(), clipKey.index));
+
+    return true;
+}
+
+bool ProcessingInteraction::removeClipData(const processing::ClipKey& clipKey, double begin, double end)
+{
+    WaveTrack* waveTrack = DomAccessor::findWaveTrack(projectRef(), TrackId(clipKey.trackId));
+    IF_ASSERT_FAILED(waveTrack) {
+        return false;
+    }
+
+    std::shared_ptr<WaveClip> clip = DomAccessor::findWaveClip(waveTrack, clipKey.index);
+    IF_ASSERT_FAILED(clip) {
+        return false;
+    }
+
+    auto initialClipStart = clip->Start();
+    auto initialClipEnd = clip->End();
```

Comment:
please do not use auto for simple types
The point here is that the compiler will easily calculate the type, but for reading a code this is unnecessary complexity, we need to know what the function returns... for reading and understanding the code, we need to see what types we are dealing with. For example, in this case, seeing that the type is double, then further in the code it is clear that there may be a problem, doubles are not safe to compare with each other (<= and =>)